snapcraft/_store.py

```diff
@@ -50,6 +50,59 @@ def _get_data_from_snap_file(snap_path):
     return snap_yaml
 
 
+def _fail_login(msg=''):
+    logger.info(msg)
+    logger.info('Login failed.')
+    return False
+
+
+def _get_url_from_error(error):
+    if error.extra:
+        return error.extra[0].get('url')
+    return None
+
+
+def _check_dev_agreement_and_namespace_statuses(store):
```

Comment:
Can we just raise exceptions if the conditions are not met. The exception would carry a friendly error message we would raise to the user just like we do for all others